convert_gvf_to_vcf/convertGVFtoVCF.py

```diff
@@ -47,75 +48,110 @@ def generate_vcf_header_unstructured_line(vcf_unstructured_key,
+        list_of_sample_information = pragma_value.split(";")
+        for sample_info in list_of_sample_information:
+            if sample_info.startswith("sample_name"):
+                sample_name = sample_info.split("=")[1]
+    return sample_name
 
-    #Go through essential pragmas
-    #TODO: list of pragmas to add:reference=file, contig, phasing,INFO#
-    list_of_pragma = ["##file-date", "##gff-version", "##gvf-version", "##species", "##genome-build"]
-    pragma_to_vcf_map = read_pragma_mapper(os.path.join(etc_folder, 'pragma_mapper.tsv'))
-    for pragma in gvf_pragmas:
-        vcf_header_key, pragma_name, pragma_value = get_pragma_name_and_value(pragma, " ", list_of_pragma, pragma_to_vcf_map)
-        pragmas_to_add.append(generate_vcf_header_unstructured_line(vcf_header_key, pragma_value))
-        # FIXME: Why are we adding header from the VCF lines
-        # for vcf_obj in list_of_vcf_objects:
-        #     pragmas_to_add.append(generate_vcf_header_unstructured_line("source", vcf_obj.source))
-    ####
-    ####
-    # Go through non-essential pragmas
-    list_of_non_essential_pragma = ["#sample", "#Study_accession", "#Study_type", "#Display_name", "#Publication"
-                                    "#Study", "#Assembly_name", "#subject"]
-    for non_essential_pragma in gvf_non_essential:
+def get_unique_sample_names(sample_names):
+    """ Takes a list of sample names and returns a unique list
+    :param: sample_names: list of sample names (can contain duplicates)
+    :return: uniq_sample_name: list of unique sample names
+    """
+    seen_sample_names = set()
+    uniq_sample_name = []
+    for sample in sample_names:
+        if sample not in seen_sample_names:
+            seen_sample_names.add(sample)
+            uniq_sample_name.append(sample)
+    return uniq_sample_name
+
+
+def convert_essential_pragmas(list_of_gvf_pragmas_to_convert,
+                              list_of_converted_pragmas,
+                              list_of_essential_pragma,
+                              pragma_to_vcf_map):
+    """ This function converts essential pragmas using the delimiter for essential pragmas (delim=" ").
+    :param: list_of_gvf_pragmas_to_convert: pragmas to be converted
+    :param: list_of_converted_pragmas: list of existing converted pragmas: will be appending results to this list
+    :param: list_of_essential_pragma: a reference list of essential pragmas
+    :param: pragma_to_vcf_map: a mapping dict of GVF pragmas and their VCF counterpart
+    : return: list_of_converted_pragmas: updated version
+    """
+    for pragma in list_of_gvf_pragmas_to_convert:
+        vcf_header_key, pragma_name, pragma_value = get_pragma_name_and_value(pragma, " ", list_of_essential_pragma, pragma_to_vcf_map)
+        list_of_converted_pragmas.append(generate_vcf_header_unstructured_line(vcf_header_key, pragma_value))
+    return list_of_converted_pragmas
+
+def convert_nonessential_pragmas(nonessential_gvf_pragmas_to_convert,
```

Comment:
The concept of `essential` vs `nonessential` pragma is confusing to me. If we're going to use this naming convention we should describe it better in the docstring. 
 - What the difference in format ?
 - What is the difference in function ?